Sources/SwiftDocC/Utility/NearMiss.swift

```diff
@@ -0,0 +1,105 @@
+            })
+        
+        // Some common prefixes result in a large number of matches. For example, many types in Swift-DocC have
+        // a "Documentation" prefix which yields a fairly high score in this implementation. To counteract this
+        // we additionally filter out any match with a score that's less than 25% of the highest match's score.
+        guard let bestScore = goodMatches.first?.score else {
+            return []
+        }
+        let matchThreshold = bestScore / 4
+        
+        return goodMatches
+            .prefix(while: { matchThreshold < $0.score })
+            // More than 10 results are likely not helpful to the user.
+            .prefix(10)
+            .map { $0.text }
+    }
+    
+    /// Computes the "score" for a collection of change segments.
+    private static func score(_ changes: CollectionChanges) -> Double {
+        // Again, there is no right or wrong way to score changes and this implementation is completely arbitrary.
+        
+        // Give the first segment a bit more weight to its contribution to the total score
+        guard let first = changes.segments.first else { return 0 }
+        var score = NearMiss.score(first) * 1.75
+        
+        for segment in changes.segments.dropFirst() {
+            score += NearMiss.score(segment)
+        }
+        return score
+    }
+        
+    /// Computes the "score" for a single collection change segments.
+    private static func score(_ segment: CollectionChanges.Segment) -> Double {
+        // Again, there is no right or wrong way to score changes and this implementation is completely arbitrary.
+        
+        // This implementation is built around a few basic ideas:
+        //
+        //  - Common segments _add_ to a change collection's score,
+        //  - Inserted and removed segments _subtract from_ a change collection's score.
+        //  - Short "common segments" occur in differences that are very different ("orange" and "lemon" both contain a "e").
+        //  - A long sequence of common elements should contribute more than an equal length sequence of different characters.
+        //    In other words; a 50% match is still "good".
+        //  - The longer a common segment is, the more "similar" to two strings are.
+        //  - A removed segment contribute more than an inserted segment (since the author had written those characters).
+        
+        switch segment.kind {
+        case .common:
+            if segment.count < 3 {
+                // 1, or 2 common characters are too few to be what a person would consider a similarity.
+                return 0.0
+            } else {
+                // To produce higher contributions for longer common sequences, this implementation sums the sequence (1...length)
+                // and adds an arbitrary constant factor.
+                return Double((1...segment.count).reduce(0, +)) + 3
```

Comment:
Nit: not sure this gets peephole optimized, but consider rewriting the this as: 
```suggestion
                return Double(((segment.count * (segment.count + 1)) / 2) + 3)
```

Reply:
You're right, this probably doesn't get optimized.

Reply:
Fixed in https://github.com/apple/swift-docc/pull/420/commits/52f57eb0dce8eb8dce145945951787eb66868329